clkmgr/pub/clkmgr/event_state.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SPDX-FileCopyrightText: Copyright © 2024 Intel Corporation.
+ */
+
+/** @file event_state.h
+ * @brief Structure for event state and event count
+ *
+ * @author Noor Azura Ahmad Tarmizi <noor.azura.ahmad.tarmizi@intel.com>
+ * @copyright © 2024 Intel Corporation.
+ *
+ */
+
+#ifndef CLKMGR_EVENT_STATE_H
+#define CLKMGR_EVENT_STATE_H
+
+#include <cstdint>
+
+#include <clkmgr/utility.h>
+
+__CLKMGR_NAMESPACE_BEGIN
+
+/**
+ * Current State for the events
+ */
```

Comment:
We need to add struct to the doxygene documentation. 
https://stackoverflow.com/questions/7199151/use-doxygen-to-document-members-of-a-c-structure-outside-of-the-structure-defini 

![image](https://github.com/user-attachments/assets/96df36ba-68bb-4c05-91fd-a7b456fe647c)

Reply:
This commit is just for moving the API from libptpmgmt pub folder back to clock manager own pub folder and make sure all necessary header is inside pub folder.  we are now WIP to refactor and rename the API according to discussion with Erez. Plan to do it on another commit, since this commit is already large